packages/internal/src/storage-test-utils.js

```diff
@@ -121,89 +131,99 @@ export const makeFakeStorageKit = (rootPath, rootOptions) => {
+          );
+        }
+        case 'set':
+        case 'setWithoutNotify': {
+          trace('toStorage set', message);
+          /** @type {StorageEntry[]} */
+          const newEntries = message.args;
+          for (const [key, value] of newEntries) {
+            if (value != null) {
+              data.set(key, value);
+            } else {
+              data.delete(key);
             }
           }
-          if (streamCell === undefined) {
-            streamCell = {
-              blockHeight: '0',
-              values: oldVal != null ? [oldVal] : [],
-            };
+          break;
+        }
+        case 'append': {
+          trace('toStorage append', message);
+          /** @type {StorageEntry[]} */
+          const newEntries = message.args;
+          for (const [key, value] of newEntries) {
+            value != null || Fail`attempt to append with no value`;
+            // In the absence of block boundaries, everything goes in a single StreamCell.
+            const oldVal = data.get(key);
+            let streamCell;
+            if (oldVal != null) {
+              try {
+                streamCell = JSON.parse(oldVal);
+                assert(isStreamCell(streamCell));
+              } catch (_err) {
+                streamCell = undefined;
+              }
+            }
+            if (streamCell === undefined) {
+              streamCell = {
+                blockHeight: '0',
+                values: oldVal != null ? [oldVal] : [],
+              };
+            }
+            streamCell.values.push(value);
+            data.set(key, JSON.stringify(streamCell));
           }
-          streamCell.values.push(value);
-          data.set(key, JSON.stringify(streamCell));
+          break;
         }
-        break;
+        case 'size':
+          // Intentionally incorrect because it counts non-child descendants,
+          // but nevertheless supports a "has children" test.
+          return [...data.keys()].filter(k =>
+            k.startsWith(`${message.args[0]}.`),
+          ).length;
+        default:
+          throw Error(`unsupported method: ${message.method}`);
       }
-      case 'size':
-        // Intentionally incorrect because it counts non-child descendants,
-        // but nevertheless supports a "has children" test.
-        return [...data.keys()].filter(k => k.startsWith(`${message.args[0]}.`))
-          .length;
-      default:
-        throw Error(`unsupported method: ${message.method}`);
-    }
-  };
-  const rootNode = makeChainStorageRoot(toStorage, rootPath, resolvedOptions);
+    },
+  });
+
+  const makeStorageNode = prepareChainStorageNode(zone);
+  const rootNode = makeChainStorageRoot(
+    storer,
+    rootPath,
+    resolvedOptions,
+    makeStorageNode,
+  );
   return {
     rootNode,
     // eslint-disable-next-line object-shorthand
     data: /** @type {Map<string, string>} */ (data),
     messages,
-    toStorage,
+    toStorage: storer.toStorage,
```

Comment:
@erights remind me, is the method of an exo (singleton) bound?

Reply:
It is not. 

See also https://github.com/Agoric/agoric-sdk/blob/0889e20ecbab673e536d0ca194e3082ceb0baf16/packages/internal/src/method-tools.js#L107 `bindAllMethods`

Reply:
@turadg given the above answer I don't understand how this could work

Reply:
Lots of failng tests. I haven't looked into them yet but probably some are due to this.

I regret requesting reviews before getting to green.